dex/src/main/scala/com/wavesplatform/dex/AddressActor.scala

```diff
@@ -266,12 +280,15 @@ class AddressActor(owner: Address,
     context.system.scheduler.scheduleOnce(settings.wsMessagesInterval, self, PrepareDiffForWsSubscribers)
   }
 
-  private def mkWsBalances(spendableBalances: Map[Asset, Long]): Map[Asset, WsBalances] = {
+  private def mkWsBalances(spendableBalances: Map[Asset, Long]): Future[Map[Asset, WsBalances]] = {
```

Comment:
Another layer of futures make things harder especially in actor. We have to think:
1. What if it fails
2. What if something happens between we run a future and receive a result

We can fill up `efc` before new assets come to AddressActor. We can do this either in the `Matcher` class, or `SpendableBalanceChanges` actor. I think it easier in `Matcher`, e.g. we can do:

```scala
  private val spendableBalancesActor = {
    actorSystem.actorOf(
      Props(
        new SpendableBalancesActor(
          wavesBlockchainAsyncClient.spendableBalances,
          wavesBlockchainAsyncClient.allAssetsSpendableBalance,
          addressActors
        )
      )
    ) unsafeTap { sba =>
      wavesBlockchainAsyncClient.spendableBalanceChanges
        .mapEval { xs =>
          Task
            .traverse(xs.valuesIterator.flatMap(_.keysIterator)) { asset =>
              Task.fromFuture(getDecimalsFromCache(asset).value)
            }
            .map(_ => xs)
        }
        .foreach(sba ! SpendableBalancesActor.Command.UpdateStates(_))(monixScheduler)
    }
  }
```

Also we could use [bufferIntrospective](https://monix.io/docs/3x/reactive/observable.html#bufferintrospective) with a big size (e.g. 100) and combine maps through monoid to improve the performance of this block because asking DEX extension could take some time.

Note this approach doesn't require `getAssetDecimals` in `AddressActor`.